temporal-serviceclient/src/main/java/io/temporal/serviceclient/TemporalSslContextBuilder.java

```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright (C) 2021 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.serviceclient;
+
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolConfig;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolNames;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContextBuilder;
+import io.grpc.netty.shaded.io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+public class TemporalSslContextBuilder {
+
+  // Default TLS protocol config that is used to communicate with TLS enabled temporal backend.
+  private static final ApplicationProtocolConfig DEFAULT_APPLICATION_PROTOCOL_CONFIG =
+      new ApplicationProtocolConfig(
+          // HTTP/2 over TLS mandates the use of ALPN to negotiate the use of the protocol.
+          ApplicationProtocolConfig.Protocol.ALPN,
+          // NO_ADVERTISE is the only mode supported by both OpenSsl and JDK providers.
+          ApplicationProtocolConfig.SelectorFailureBehavior.NO_ADVERTISE,
+          // ACCEPT is the only mode supported by both OpenSsl and JDK providers.
+          ApplicationProtocolConfig.SelectedListenerFailureBehavior.ACCEPT,
+          // gRPC requires http2 protocol.
+          ApplicationProtocolNames.HTTP_2);
+
+  private final InputStream keyCertChainInputStream;
+  private final InputStream keyInputStream;
+  private TrustManager trustManager;
+  private boolean useInsecureTrustManager;
+  private String keyPassword;
+
+  /**
+   * @param keyCertChainInputStream - an input stream for an X.509 client certificate chain in PEM
+   *     format.
+   * @param keyInputStream - an input stream for a PKCS#8 client private key in PEM format.
+   */
+  public TemporalSslContextBuilder(
+      InputStream keyCertChainInputStream, InputStream keyInputStream) {
+    this.keyCertChainInputStream = keyCertChainInputStream;
+    this.keyInputStream = keyInputStream;
+  }
```

Comment:
I would create a static method and hide the constructor:

TemporalSslContextBuilder.newBuilder(cert, key)

I would also rename the parameters. Having "InputStream" postfix doesn't provide any value.

Reply:
Done